distributor/distributor.go

```diff
@@ -320,6 +338,19 @@ func (d *Distributor) Push(ctx context.Context, req *remote.WriteRequest) (*cort
 	return &cortex.WriteResponse{}, nil
 }
 
+func (d *Distributor) getOrCreateIngestLimiter(userID string) *rate.Limiter {
+	d.ingestLimitersMtx.Lock()
```

Comment:
Note I'm intentionally just going for a normal mutex here because I figured an RWLock didn't really matter in this path (access once per HTTP request).